app/models/mixins/archived_mixin.rb

```diff
@@ -16,6 +16,24 @@ def active?
     deleted_on.nil?
   end
 
+  module ClassMethods
+    def were_active_at_time(time)
```

Comment:
These `were_active_at_time` and `walk_history` don't belong in this PR, I'm just sketching what kind of history access we might want in the future...
I'll remove them when PR is ready (so brakeman travis failure on this will go away)